app/src/main/java/com/devshish/internship/presentation/ui/web/WebFragment.kt

```diff
@@ -0,0 +1,56 @@
+package com.devshish.internship.presentation.ui.web
+
+import android.os.Build
+import android.os.Bundle
+import android.view.View
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import android.widget.Toast
+import androidx.annotation.RequiresApi
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.devshish.internship.R
+import com.devshish.internship.databinding.FragmentWebBinding
+import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class WebFragment : Fragment(R.layout.fragment_web) {
+
+    private val binding by viewBinding(FragmentWebBinding::bind)
+    private val viewModel: WebViewModel by viewModel()
+
+    private val args: WebFragmentArgs by navArgs()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val client = object : WebViewClient() {
+            @RequiresApi(Build.VERSION_CODES.N)
+            override fun shouldOverrideUrlLoading(
+                view: WebView?,
+                request: WebResourceRequest?
+            ): Boolean =
+                if (request?.isRedirect == true) {
+                    viewModel.checkUrl(request.url)
+                    true
+                } else false
+        }
```

Comment:
1. Remove @RequiresApi - because it will not work with Android versions below apiVersion=24 (N)
2. I think you can safely remove isRedirect and just checkUrl on viewModel  (so it will be fun `checkUrl(url: Uri?): Boolean`)
3. You should not forget about **state** value you passed - it is needed so you can check that current url is 100% correct and there are not just some random "code" param (so you need to perform addition check for **state**)
<img width="319" alt="image" src="https://user-images.githubusercontent.com/23121846/136078007-3b33e8b7-1b0f-4aa0-a606-9df9e22b6243.png">
4. move Client to Fragment field (Like on screenshot)
<img width="593" alt="image" src="https://user-images.githubusercontent.com/23121846/136078421-fa505aab-7dc6-4a0c-8546-78a7358e8fc0.png">

Reply:
Move WebViewClient to separate class and file